src/compiler/checker.ts

```diff
@@ -27609,7 +27609,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 location = location.parent;
             }
             if (isExpressionNode(location) && (!isAssignmentTarget(location) || isWriteAccess(location))) {
-                const type = getTypeOfExpression(location as Expression);
+                const type = removeOptionalTypeMarker(getTypeOfExpression(location as Expression));

```

Comment:
I'm not that familiar with this part of the code. Why doesn't this also apply to the entirety of `a?.b.c`?

Reply:
The "optional type" marker is not added at the outermost `PropertyAccessExpression`, only for inner optional chain parts. At the outermost `PropertyAccessExpression`, we just union it with the normal `undefined` type.

Reply:
See `propagateOptionalTypeMarker` in checker.ts for the logic related to when the marker type is added vs a regular `undefined`.